mobile/src/employee-details/employee-details.tsx

```diff
@@ -166,6 +174,27 @@ export class EmployeeDetailsImpl extends Component<EmployeeDetailsProps & Employ
         );
     }
 
+    private renderDaysCounters = (employee: Employee) => {
+        const allVacationDaysCounter = new VacationDaysCounter(employee.vacationDaysLeft);
+
+        const daysConverter = new ConvertHoursCreditToDays();
+        const calculatedDays = daysConverter.convert(employee.hoursCredit);
+
+        const hoursCreditCounter = new HoursCreditCounter(employee.hoursCredit, calculatedDays.days, calculatedDays.rest);
+
+        const counters: DaysCountersModel = {
+            allVacationDays: allVacationDaysCounter,
+            hoursCredit: hoursCreditCounter,
+        };
+
+        if (employee.vacationDaysLeft !== null || employee.hoursCredit !== null) {
```

Comment:
I would do this condition checking at the very beginning, before creating counters components.